File: SovietTransistorsDatabase/Domain/Cyrillic.cs
namespace SovietTransistors.Domain;

public static class Cyrillic
{
    /// <summary>Заглавная буква русского алфавита (А–Я, Ё).</summary>
    public static bool IsUpperLetter(char c) => (c >= 'А' && c <= 'Я') || c == 'Ё';

    /// <summary>Строка из одной или двух заглавных русских букв.</summary>
    public static bool IsUpperLetters(string? s) =>
        !string.IsNullOrEmpty(s) && s.Length <= 2 && s.All(IsUpperLetter);
}

File: internal/domain/cyrillic.go
package domain

// IsUpperLetter — заглавная буква русского алфавита (А–Я, Ё).
func IsUpperLetter(c rune) bool {
	return (c >= 'А' && c <= 'Я') || c == 'Ё'
}

// IsUpperLetters — строка из одной или двух заглавных русских букв.
func IsUpperLetters(s string) bool {
	if s == "" {
		return false
	}
	letters := []rune(s)
	if len(letters) > 2 {
		return false
	}
	for _, c := range letters {
		if !IsUpperLetter(c) {
			return false
		}
	}
	return true
}
